tedana/datasets/fetchers.py

```diff
@@ -0,0 +1,246 @@
+    for each subject in the Cambridge dataset.
+    Files are downloaded from Open Science Framework (OSF).
+    For more information on the data and its preprocessing, see:
+    https://osf.io/9wcb8/
+
+    Parameters
+    ----------
+    n_subjects : int
+        The number of subjects to load. If None, all the subjects are
+        loaded. Total 88 subjects.
+    data_dir : str
+        Path of the data directory. Used to force data storage in a specified
+        location. If None is given, data are stored in home directory.
+    url : str
+        Override download URL. Used for test only (or if you setup a mirror of
+        the data).
+    resume : bool
+        Whether to resume download of a partly-downloaded file.
+    verbose : int
+        Defines the level of verbosity of the output.
+    Returns
+    -------
+    func : list of str (Nifti files)
+        Paths to functional MRI data (4D) for each subject.
+    """
+    dataset_name = 'cambridge'
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir,
+                                verbose=verbose)
+
+    if url is None:
+        # Download from the relevant OSF project, using hashes generated
+        # from the OSF API. Note the trailing slash. For more info, see:
+        # https://gist.github.com/emdupre/3cb4d564511d495ea6bf89c6a577da74
+        url = 'https://osf.io/download/{}/'
+
+    func = '{0}_task-rest_{1}_space-scanner_desc-partialPreproc_bold.nii.gz'
+
+    # The gzip contains unique download keys per Nifti file and confound
+    # pre-extracted from OSF. Required for downloading files.
+    package_directory = os.path.dirname(os.path.abspath(__file__))
+    dtype = [('participant_id', 'U12'), ('echo_id', 'U12'),
+             ('key_bold', 'U24')]
+    names = ['participant_id', 'echo_id', 'key_b']
+    # csv file contains download information related to OpenScience(osf)
+    osf_data = csv_to_array(os.path.join(package_directory, "data",
+                                         "cambridge_echos.csv"),
+                            skip_header=True, dtype=dtype, names=names)
+    funcs = []
+    participant_id, echo_id, uuid = zip(*osf_data)
+    participants = np.unique(participant_id)[:n_subjects]
+
+    for participant_id in participants:
+        this_osf_id = osf_data[osf_data['participant_id'] == participant_id]
+        participant_funcs = []
+
+        for entry in this_osf_id:
+            echo_id = entry['echo_id']
+            # Download bold images for each echo
+            func_url = url.format(entry['key_b'])
+            func_file = [(func.format(participant_id, echo_id),
+                          func_url,
+                          {'move': func.format(participant_id, echo_id)})]
+            path_to_func = _fetch_files(data_dir, func_file, resume=resume,
+                                        verbose=verbose)[0]
+            participant_funcs.append(path_to_func)
+        funcs.append[tuple(participant_funcs)]
```

Comment:
```suggestion
        funcs.append(tuple(participant_funcs))
```

This should fix the bug @jbteves noted.